aws/resource_aws_sagemaker_image.go

```diff
@@ -0,0 +1,213 @@
+				),
+			},
+			"role_arn": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateArn,
+			},
+			"display_name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringLenBetween(1, 128),
+			},
+			"description": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringLenBetween(1, 512),
+			},
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceAwsSagemakerImageCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).sagemakerconn
+
+	name := d.Get("image_name").(string)
+	input := &sagemaker.CreateImageInput{
+		ImageName: aws.String(name),
+		RoleArn:   aws.String(d.Get("role_arn").(string)),
+	}
+
+	if v, ok := d.GetOk("display_name"); ok {
+		input.DisplayName = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("description"); ok {
+		input.Description = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("tags"); ok {
+		input.Tags = keyvaluetags.New(v.(map[string]interface{})).IgnoreAws().SagemakerTags()
+	}
+
+	log.Printf("[DEBUG] sagemaker Image create config: %#v", *input)
+	_, err := conn.CreateImage(input)
+	if err != nil {
+		return fmt.Errorf("error creating SageMaker Image: %w", err)
+	}
+
+	d.SetId(name)
+
+	if _, err := waiter.ImageCreated(conn, d.Id()); err != nil {
+		return fmt.Errorf("error waiting for SageMaker Image (%s) to create: %w", d.Id(), err)
+	}
+
+	return resourceAwsSagemakerImageRead(d, meta)
+}
+
+func resourceAwsSagemakerImageRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).sagemakerconn
+	ignoreTagsConfig := meta.(*AWSClient).IgnoreTagsConfig
+
+	image, err := finder.ImageByName(conn, d.Id())
+	if err != nil {
+		if isAWSErr(err, sagemaker.ErrCodeResourceNotFound, "No Image with the name") {
+			d.SetId("")
+			log.Printf("[WARN] Unable to find SageMaker Image (%s); removing from state", d.Id())
+			return nil
+		}
+		return fmt.Errorf("error reading SageMaker Image (%s): %w", d.Id(), err)
+
+	}
+
+	arn := aws.StringValue(image.ImageArn)
+	d.Set("image_name", image.ImageName)
+	d.Set("arn", arn)
+	d.Set("role_arn", image.RoleArn)
+	d.Set("display_name", image.DisplayName)
+	d.Set("description", image.Description)
+
+	tags, err := keyvaluetags.SagemakerListTags(conn, arn)
+
+	if err != nil {
+		return fmt.Errorf("error listing tags for SageMaker Image (%s): %w", d.Id(), err)
+	}
+
+	if err := d.Set("tags", tags.IgnoreAws().IgnoreConfig(ignoreTagsConfig).Map()); err != nil {
+		return fmt.Errorf("error setting tags: %w", err)
+	}
+
+	return nil
+}
+
+func resourceAwsSagemakerImageUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).sagemakerconn
+	needsUpdate := false
+
+	input := &sagemaker.UpdateImageInput{
+		ImageName: aws.String(d.Id()),
+	}
+
+	var deleteProperties []*string
+
+	if d.HasChange("role_arn") {
+		input.RoleArn= aws.String(d.Get("role_arn").(string))
+	}
```

Comment:
Fix formatting.

```suggestion
	if d.HasChange("role_arn") {
		input.RoleArn = aws.String(d.Get("role_arn").(string))
	}
```

Reply:
ill add a test for role change, missed that.